pipelines/collate_plots.py

```diff
@@ -30,261 +26,65 @@ def merge_pdfs_and_save(
-
-    target_filenames = ensure_listlike(target_filenames)
-
-    forecast_dirs = get_all_forecast_dirs(model_base_dir, diseases=disease)
-
-    # first collate locations for a given date
-    logger.info(
-        "Collating plots across locations, by forecast date. "
-        f"{len(forecast_dirs)} dates to process."
-    )
-    for f_dir in forecast_dirs:
-        logger.info(f"Collating plots from {f_dir}")
-        process_dir(
-            base_dir=Path(model_base_dir, f_dir, "model_runs"),
-            target_filenames=target_filenames,
-            save_dir=Path(model_base_dir, f_dir),
-        )
-    logger.info("Done collating across locations by date.")
-
-    # then collate dates, adding the disease name
-    # as a prefix for disambiguation since the
-    # top-level directory may contain forecasts
-    # for multiple diseases.
-    logger.info("Collating plots from forecast date directories...")
-    process_dir(
-        base_dir=model_base_dir,
-        target_filenames=target_filenames,
-        save_dir=save_dir,
-        file_prefix=f"{disease}_",
-        subdirs_only=forecast_dirs,
-    )
-
-    logger.info("Done collating plots from forecast date directories.")
-
-    return None
-
-
-def main(
-    dir_of_forecast_dirs: str | Path,
-    single_forecast_dir: str | Path,
-    target_filenames: list[str],
-    disease: str = None,
-) -> None:
-    if not ((dir_of_forecast_dirs is None) ^ (single_forecast_dir is None)):
-        raise ValueError(
-            "Must provide exactly one of "
-            "'dir_of_forecast_dirs' (to process multiple "
-            "groups of forecasts) or "
-            "'single_forecast_dir' "
-            "(to process a single set of forecasts"
-        )
-    elif dir_of_forecast_dirs is not None:
-        if disease is None:
-            raise ValueError(
-                "'disease' must not be None when collating plots "
-                "from multiple forecast subdirectories"
-            )
-        collate_from_all_subdirs(
-            dir_of_forecast_dirs, disease, target_filenames
-        )
-    elif single_forecast_dir is not None:
-        process_dir(single_forecast_dir, target_filenames)
-    return None
+    for model, file_dict in pdf_groups.items():
+        for original_file_name, pdf_list in file_dict.items():
+            if len(pdf_list) > 1:
+                output_filename = f"{model}-{original_file_name}"
+                output_path = model_batch_dir / "figures" / output_filename
+                output_path.parent.mkdir(parents=True, exist_ok=True)
+                merge_pdfs(pdf_list, output_path)
+            else:
+                print(
+                    f"Skipping {original_file_name} in {model}, only one file found."
+                )
 
 
 parser = argparse.ArgumentParser(
     description=("Collate forecast plots from subdirectories into single PDFs")
 )
 
 parser.add_argument(
-    "--dir-of-forecast-dirs",
+    "modelbatchdir",
```

Comment:
weak preference for separating out

```suggestion
    "model_batch_dir",
```

Reply:
https://github.com/CDCgov/pyrenew-hew/pull/287/commits/43a15c4243329362c99cd5a1b34c6c66720e51ce